cmd/thanos/sidecar.go

```diff
@@ -53,6 +55,8 @@ func registerSidecar(m map[string]setupFunc, app *kingpin.Application, name stri
 
 	uploadCompacted := cmd.Flag("shipper.upload-compacted", "[Experimental] If true sidecar will try to upload compacted blocks as well. Useful for migration purposes. Works only if compaction is disabled on Prometheus.").Default("false").Hidden().Bool()
 
+	validateProm := cmd.Flag("sidecar.validate-prom", "[Experimental]If true sidecar will check Prometheus' flags to ensure disabled compaction and 2h block-time.").Default("true").Hidden().Bool()
```

Comment:
Should be experimental flag enabled by default?

Reply:
IMHO it shouldn't even be a flag nor experimental since we don't support other block lengths besides 2h

Reply:
IMHO it shouldn't even be a flag nor experimental since we don't support other configurations which this checks. However, it might be useful to disable it for debugging.